scripts/backout_data_collection.py

```diff
@@ -0,0 +1,172 @@
+import json
+import logging
+import os
+
+from tqdm import tqdm
+
+from bugbug import bugzilla, db, repository
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def download_databases() -> None:
+    logger.info("Downloading bugs database...")
+    assert db.download(bugzilla.BUGS_DB)
+
+    logger.info("Downloading commits database...")
+    assert db.download(repository.COMMITS_DB, support_files_too=True)
+
+
+def preprocess_commits_and_bugs() -> tuple[dict, dict, dict]:
+    logger.info("Preprocessing commits and bugs...")
+    commit_dict = {}
+    bug_to_commit_dict = {}
+
+    # store commits with their hashes and bug IDs as keys
+    for commit in tqdm(
+        repository.get_commits(
+            include_no_bug=True, include_backouts=True, include_ignored=True
+        ),
+        desc="Processing commits",
+    ):
+        commit_dict[commit["node"]] = {
+            "node": commit["node"],
+            "bug_id": commit["bug_id"],
+            "desc": commit["desc"],
+            "pushdate": commit["pushdate"],
+            "backedoutby": commit["backedoutby"],
+            "backsout": commit["backsout"],
+        }
+
+        if commit_dict[commit["node"]]["bug_id"] not in bug_to_commit_dict:
+            bug_to_commit_dict[commit["bug_id"]] = [commit_dict[commit["node"]]]
+        else:
+            bug_to_commit_dict[commit["bug_id"]].append(commit_dict[commit["node"]])
+
+    logger.info("Preprocessing bugs")
+    bug_dict = {}
+
+    num_lines = sum(1 for line in open(bugzilla.BUGS_DB, "r"))
```

Comment:
I don't think it is worth iterating over the whole dataset to count the bugs. We could use `tqdm` without the total parameter.